### PR TITLE
🐛Fix Settings Update Navigate

### LINKED ISSUE #1185: 🐛Navigate after settings update broken
#### Describe your issue

Currently the config-panel navigates the user back to where he came from.
This makes the diagram-detail view open a diagram on a PE that BPMN-Studio is eventually not connected to.

#### How can others reproduce the issue?

1. Open a diagram.
2. Go to settings.
3. Change PE Route.
4. Click Save.

#### Possible solution

Deactivate the current solution after a change in settings.
Let diagram-detail check if there is an active Solution, decide whether or not it stays open.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by Vyperus**
**Changes:**

1. Adds a check to determine if the active solution is remote in config-panel.
2. Adds a check for undefined active solution in diagram-detail.

**Issues:**

Closes #1185 

PR: #1186

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/config-panel/config-panel.ts [hunk 1]
```diff
@@ -3,8 +3,10 @@ import {bindable, computedFrom, inject} from 'aurelia-framework';
 import {OpenIdConnect} from 'aurelia-open-id-connect';
 import {Router} from 'aurelia-router';
 
+import { IDiagram } from '@process-engine/solutionexplorer.contracts';
+import { runInThisContext } from 'vm';
```
Comment (Paulomart) [inline src/modules/config-panel/config-panel.ts]: I think this was not intended

File: src/modules/design/diagram-detail/diagram-detail.ts [hunk 2]
```diff
@@ -86,6 +86,13 @@ export class DiagramDetail {
     }
 
     this._activeSolutionEntry = await this._solutionService.getActiveSolutionEntry();
+
+    if (this._activeSolutionEntry === undefined) {
```
Comment (SteffenKn) [inline src/modules/design/diagram-detail/diagram-detail.ts]:
```suggestion
    const activeSolutionEntryDoesNotExist: boolean = this._activeSolutionEntry === undefined;
    if (activeSolutionEntryDoesNotExist) {
```

File: src/modules/config-panel/config-panel.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import {OpenIdConnect} from 'aurelia-open-id-connect';
 import {Router} from 'aurelia-router';
 
 import {IAuthenticationService} from '../../contracts/authentication/IAuthenticationService';
-import {AuthenticationStateEvent, ISolutionService, NotificationType} from '../../contracts/index';
+import {AuthenticationStateEvent, ISolutionEntry, ISolutionService, NotificationType} from '../../contracts/index';
```
Comment (alexanderkasten) [inline src/modules/config-panel/config-panel.ts]:
```suggestion
import {
  AuthenticationStateEvent,
  ISolutionEntry,
  ISolutionService,
  NotificationType
} from '../../contracts/index';
```

File: src/modules/config-panel/config-panel.ts [hunk 4]
```diff
@@ -4,7 +4,10 @@ import {OpenIdConnect} from 'aurelia-open-id-connect';
 import {Router} from 'aurelia-router';
 
 import {IAuthenticationService} from '../../contracts/authentication/IAuthenticationService';
-import {AuthenticationStateEvent, ISolutionService, NotificationType} from '../../contracts/index';
+import {AuthenticationStateEvent,
```
Comment (alexanderkasten) [inline src/modules/config-panel/config-panel.ts]: Wrong indentation, please do it like in all other .ts files with two spaces und a newline after a bracket and a newline before the closing bracket.

Comment (SteffenKn) [reply src/modules/config-panel/config-panel.ts]:
```suggestion
import {
  AuthenticationStateEvent,
  ISolutionEntry,
  ISolutionService,
  NotificationType,
} from '../../contracts/index';
```

**Comment by SteffenKn** [comment]
If I now work on a deployed diagram and, for example, change the authority URL or simply click Save without changing anything, I am redirected to the start page.

File: src/modules/config-panel/config-panel.ts (hunk 4 above)
Comment (Vyperus) [reply src/modules/config-panel/config-panel.ts]: We do it both ways! I adjusted it for now but we seriously need some style guide for this kind of stuff.

File: src/modules/config-panel/config-panel.ts [hunk 5]
```diff
@@ -158,4 +150,43 @@ export class ConfigPanel {
...
+    const baseRouteIsInternalProcessEngine: boolean = this.baseRoute === window.localStorage.getItem('InternalProcessEngineRoute');
+    if (baseRouteIsInternalProcessEngine) {
+      window.localStorage.setItem('processEngineRoute', '');
+    } else {
+      window.localStorage.setItem('processEngineRoute', this.baseRoute);
+    }
+  }
+
+  private _updateAuthority(): void {
+    const authorityIsSet: boolean = this.authority !== undefined
```
Comment (SteffenKn) [inline src/modules/config-panel/config-panel.ts]: Please fix the indentation, like so:

```
    const authorityIsSet: boolean = this.authority !== undefined
    							 && this.authority !== null
   							     && this.authority !== '';
```

**Pull request merged**